administrator/components/com_associations/Model/AssociationsModel.php

```diff
@@ -432,6 +445,63 @@ protected function getListQuery()
 			}
 		}
 
+		// Filter by association state
+		$assocStateField = $this->state->get('assocstate');
+
+		if ($assocStateField !== 'all')
+		{
+			// Not associated
+			if ($assocStateField === 'not_associated')
+			{
+				$languageQuery = $db->getQuery(true)
+					->select('COUNT(*)')
+					->from($db->quoteName('#__languages'));
+				$db->setQuery($languageQuery);
+				$countLanguages = $db->loadResult();
+
+				// Get all keys where not all languages are associated.
+				$assocQuery = $db->getQuery(true)
+					->select($db->quoteName('key'))
+					->from($db->quoteName('#__associations'))
+					->group($db->quoteName('key'))
+					->having('COUNT(*) < ' . $countLanguages);
+
+				// Join over associations where id does not exist
+				$query->where('((' . $db->quoteName('asso.id') . ' IS NULL )'
+					// Or if we are on the child language and there is no parent
+					. ' OR ( ' . $db->quoteName('asso2.parent_id') . ' = ' . $db->quote('-1') . ')'
```

Comment:
You don't need to quote integer constants. Simple add it to the query.
```suggestion
					. ' OR ( ' . $db->quoteName('asso2.parent_id') . ' = -1)'
```